### PR TITLE
Add t23Formatter

### PR DESCRIPTION
**Pull request by ksumman**

### CODE REVIEW

File: lib/xcpretty/formatters/t23Formatter.rb [hunk 1]
```diff
@@ -0,0 +1,99 @@
+module XCPretty
+    class T23Formatter < Simple
+        def format_t23_output(text)
```
Comment (pbrobst23) [inline lib/xcpretty/formatters/t23Formatter.rb:5]: spec/xcpretty/formatters/formatter_spec.rb needs a case to test this behavior

File: lib/xcpretty/parser.rb [hunk 2]
```diff
@@ -4,6 +4,8 @@ module XCPretty
 
   module Matchers
 
+    T23_MATCHER = /T23: */
```
Comment (pbrobst23) [inline lib/xcpretty/parser.rb]: spec/xcpretty/parser_spec.rb needs a case to test this behavior

**Comment by ksumman** [comment]
Tests are added. @pbrobst23  please review

File: lib/xcpretty/formatters/t23Formatter.rb [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            INDENT + INDENT + text[4..-1] #Ignore the T23: prefix
```
Comment (pbrobst23) [inline lib/xcpretty/formatters/t23Formatter.rb]: This 4 is driven by the length of the prefix (also seen inside T23_MATCHER below). 

Can you pull the prefix into a constant and share it between them?

File: lib/xcpretty/formatters/t23Formatter.rb [hunk 4]
```diff
@@ -0,0 +1,102 @@
+module XCPretty
+    class T23Formatter < Simple
+        T23_CONSOLE_OUTPUT_PREFIX = "T23:"
+
+        def format_t23_output(text)
+            t23PrefixLenght = T23_CONSOLE_OUTPUT_PREFIX.length
```
Comment (pbrobst23) [inline lib/xcpretty/formatters/t23Formatter.rb]: t23PrefixLength (spelling)

**Comment by pbrobst23** [comment]
Approved after spelling fix

**Pull request merged**